oauthproxy.go

```diff
@@ -885,6 +900,12 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 		} else {
 			req.Header.Del("X-Forwarded-Preferred-Username")
 		}
+
+		if len(session.Groups) > 0 {
+			req.Header["X-Forwarded-Groups"] = []string{p.encodeArrayHeader(session.Groups)}
+		} else {
+			req.Header.Del("X-Forwarded-Groups")
+		}
```

Comment:
Tempted to suggest we make this a loop instead and let the req header code perform the join

```suggestion
		req.Header.Del("X-Forwarded-Groups")
		for _, group := range session.Groups {
			req.Header.Add("X-Forwarded-Groups", group)
		}
```